eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -170,37 +170,11 @@ jobs:
           goals: 'install'
 
       - task: Maven@3
-        displayName: 'Install all client libraries (for SpotBugs analysis)'
+        displayName: 'Install Module and Run SpotBugs, Checkstyle, RevApi, and Javadoc'
         inputs:
-          mavenPomFile: pom.client.xml
-          options: '$(DefaultOptions) -DskipTests -Dgpg.skip -Dmaven.javadoc.skip=true -Dcheckstyle.skip=true -Dspotbugs.skip=true -Drevapi.skip=true -Dverify-readme'
-          mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
-          javaHomeOption: 'JDKVersion'
-          jdkVersionOption: '1.11'
-          jdkArchitectureOption: 'x64'
-          publishJUnitResults: false
-          goals: 'install'
-
-      # Run SpotBugs, CheckStyle, Revapi, and JavaDoc tools to ensure high code quality - these will fail the build if issues are found
-      - task: Maven@3
-        displayName: 'Run SpotBugs, CheckStyle, Revapi, and JavaDoc'
-        inputs:
-          mavenPomFile: pom.client.xml
-          options: '-DskipTests -Dgpg.skip'
-          mavenOptions: ''
-          javaHomeOption: 'JDKVersion'
-          jdkVersionOption: '1.11'
-          jdkArchitectureOption: 'x64'
-          publishJUnitResults: false
-          goals: 'spotbugs:check checkstyle:checkstyle-aggregate revapi:check javadoc:aggregate'
-
-      # We run a separate SpotBugs aggregate report step here to roll-up all the issues identified per-module
-      - task: Maven@3
-        displayName: 'Run SpotBugs aggregate analysis'
-        inputs:
-          mavenPomFile: eng/spotbugs-aggregate-report/pom.xml
-          options: '--batch-mode -DskipTests -Dgpg.skip'
-          mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
+          mavenPomFile: sdk/${{parameters.ServiceDirectory}}/pom.service.xml
```

Comment:
This looks promising. Are you sure that running the install goal on the aggregate runs what you are expecting here?

Reply:
Install runs slightly modified steps in checkstyle:check and javadoc:javadoc, these don't generate aggregates. Given that this step is ran during all PR validations I think this would be fine given that the nightly run will still run the site:site site:stage steps in reporting which are aggregates.

Any concerns with PR validation slightly changing by not running aggregates?

Reply:
Just coming back from being on leave. I don't have a concern about not running aggregate reports during PR validation. Ideally I'd like to eliminate aggregate report generation out of all pipelines.